src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -184,37 +184,40 @@ public void DrawGeometry(IBrush brush, IPen pen, IGeometryImpl geometry)
         }
 
         /// <inheritdoc />
-        public void DrawRectangle(IPen pen, Rect rect, float cornerRadius = 0)
+        public void DrawRectangle(IBrush brush, IPen pen, Rect rect, double radiusX, double radiusY)
         {
-            using (var paint = CreatePaint(pen, rect.Size))
-            {
-                var rc = rect.ToSKRect();
+            var rc = rect.ToSKRect();
+            var isRounded = Math.Abs(radiusX) > double.Epsilon || Math.Abs(radiusX) > double.Epsilon;
```

Comment:
I believe here you meant to check radiusX and radiusY, but you checked radiusX twice.

Reply:
Same problem in `Windows\Avalonia.Direct2D1\Media\DrawingContextImpl.cs`

Reply:
Y. Will fix asap.

Reply:
#3324

Reply:
Sharp eyes @Kermalis ! 👀